src/nvim/path.c

```diff
@@ -2186,9 +2186,15 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
 /// Check if the given file is absolute.
 ///
-/// This just checks if the file name starts with '/' or '~'.
 /// @return `TRUE` if "fname" is absolute.
 int path_is_absolute_path(const char_u *fname)
 {
+#ifdef WIN32
+  // A name like "d:/foo" and "//server/share" is absolute
+  return ((fname[0] && fname[1] == ':' && (fname[2] == '/' || fname[2] == '\\'))
```

Comment:
In `get_past_head()` we used `isalpha(fname[0])` instead of `fname[0]` to check for `'d'` in `"d:"`. Why is it not used here?